app/assets/javascripts/autocomplete.js

```diff
@@ -1,42 +1,16 @@
-+function ($) {
-  var Autocomplete = (function () {
-    function Autocomplete (element) {
-      var self = this
++function () {
+  $(document).on('change', 'input[data-autocomplete-url]', function () {
+    var $input = $(this)
 
-      self.element       = element
-      self.targetElement = $(element.data('autocompleteTarget'))
-
-      self.init()
-    }
-
-    Autocomplete.prototype.init = function () {
-      var self = this
-
-      self.element.autocomplete({
-        type:      'get',
-        minLength: self.element.data('autocompleteMinLength'),
-        source:    self._source.bind(self),
-        select:    self._selected.bind(self)
-      })
-
-      self._markAsObserved()
-      self._listenChanges()
-    }
-
-    Autocomplete.prototype._markAsObserved = function () {
-      this.element.attr('data-observed', true)
-    }
-
-    Autocomplete.prototype._listenChanges = function () {
-      var self = this
+    if (!$input.val().trim())
```

Comment:
Me acabo de dar cuenta que en el otro PR lo dejé pasar, pero prefiero el estilo:

```suggestion
    if (! $input.val().trim())
```

Siempre que no chille el lintern, y si chilla hay que enseñarle que no =P